src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -528,6 +556,48 @@ private void SetProxyOptions(SafeCurlHandle requestHandle, Uri requestUri)
             }
         }
 
+        private void SetRequestHandleCredentialsOptions(SafeCurlHandle requestHandle, RequestCompletionSource state)
+        {
+            NetworkCredential credentials = GetNetworkCredentials(state.Handler._serverCredentials, state.RequestMessage.RequestUri);
+            if (credentials != null)
+            {
+                if (string.IsNullOrEmpty(credentials.UserName))
+                {
+                    throw new ArgumentException(SR.net_http_argument_empty_string, "UserName");
```

Comment:
Is this when the exception occurs in WinHttpHandler, rather than earlier when the credential is actually provided?